packages/assistify-deepl-translation/server/deeplTranslate.js

```diff
@@ -0,0 +1,183 @@
+	 * @param {object} message
+	 * @param {object} targetLanguages
+	 * @returns {object} translations: Translated messages for each language
+	 */
+	_sendRequestTranslateMessage(message, targetLanguages) {
+		const translations = {};
+		let msgs = message.msg.split('\n');
+		msgs = msgs.map(msg => encodeURIComponent(msg));
+		const query = `text=${ msgs.join('&text=') }`;
+		const supportedLanguages = this._getSupportedLanguages('en');
+		targetLanguages.forEach(language => {
+			if (language.indexOf('-') !== -1 && !_.findWhere(supportedLanguages, {language})) {
+				language = language.substr(0, 2);
+			}
+			try {
+				const result = HTTP.get(this.apiEndPointUrl, {
+					params: {
+						auth_key: this.apiKey,
+						target_lang: language
+					}, query
+				});
+
+				if (result.statusCode === 200 && result.data && result.data.translations && Array.isArray(result.data.translations) && result.data.translations.length > 0) {
+					// store translation only when the source and target language are different.
+					if (result.data.translations.map(translation => translation.detected_source_language).join() !== language) {
+						const txt = result.data.translations.map(translation => translation.text).join('\n');
+						translations[language] = this.deTokenize(Object.assign({}, message, {msg: txt}));
+					}
+				}
+			} catch (e) {
+				SystemLogger.error('Error translating message', e);
+			}
+		});
+		return translations;
+	}
+
+	/**
+	 * Returns translated message attachment description in target languages.
+	 * @private
+	 * @param {object} attachment
+	 * @param {object} targetLanguages
+	 * @returns {object} translated messages for each target language
+	 */
+	_sendRequestTranslateMessageAttachments(attachment, targetLanguages) {
+		const translations = {};
+		const query = `text=${ encodeURIComponent(attachment.description || attachment.text) }`;
+		const supportedLanguages = this._getSupportedLanguages('en');
+		targetLanguages.forEach(language => {
+			if (language.indexOf('-') !== -1 && !_.findWhere(supportedLanguages, {language})) {
+				language = language.substr(0, 2);
+			}
+			try {
+				const result = HTTP.get(this.apiEndPointUrl, {
+					params: {
+						auth_key: this.apiKey,
+						target_lang: language
+					}, query
+				});
+				if (result.statusCode === 200 && result.data && result.data.translations && Array.isArray(result.data.translations) && result.data.translations.length > 0) {
+					if (result.data.translations.map(translation => translation.detected_source_language).join() !== language) {
+						translations[language] = result.data.translations.map(translation => translation.text).join('\n');
+					}
+				}
+			} catch (e) {
+				SystemLogger.error('Error translating message attachment', e);
+			}
+		});
+		return translations;
+	}
+}
+
+
+Meteor.startup(() => {
+	TranslationProviderRegistry.registerProvider(new DeeplAutoTranslate());
+	RocketChat.AutoTranslate = TranslationProviderRegistry.getActiveServiceProvider();
```

Comment:
why do you do this in all providers? Should there not be a single hook for that?